src/main/java/com/pinterest/secor/common/SecorSchemaRegistryClient.java

```diff
@@ -67,10 +70,27 @@ public GenericRecord deserialize(String topic, byte[] message) {
         return record;
     }
 
+    /**
+     * Get Avro schema of a topic. It uses the cache that either is set by calling {@link #deserialize(String, byte[])}
+     * or querying this method to avoid hitting Schema Registry for each call.
+     * It uses standard "subject name" strategy and it is topic_name-value.
+     *
+     * @param topic a Kafka topic to query the schema for
+     * @return Schema object for the topic
+     * @throws IllegalStateException if there is no schema registered for this topic or it is not able to fetch it
+     */
     public Schema getSchema(String topic) {
         Schema schema = schemas.get(topic);
         if (schema == null) {
-            throw new IllegalStateException("Avro schema not found for topic " + topic);
+            try {
+                SchemaMetadata schemaMetadata = schemaRegistryClient.getLatestSchemaMetadata(topic + "-value");
```

Comment:
This seems odd to me.  Who defines that convention: 'topic_name-value'?

Reply:
basically, there are three strategies configurable in Kafka to work with Schema Registry explained [here](https://docs.confluent.io/current/schema-registry/serializer-formatter.html#sr-avro-subject-name-strategy). The default one which most of the deployments use is "TopicNameStrategy". In this strategy

- The subject name for the value is "topic-value"
- The subject name for the key is "key-value"

Reply:
is getSchema() only interested in value schema (or key schema as well)?

Reply:
Only value schema. Key is only an internal Kafka thing and is not gonna be stored in the Avro/AvroParquet files